web/src/compose_validate.js

```diff
@@ -557,6 +558,14 @@ function validate_private_message() {
     return true;
 }
 
+export function check_uploads_in_progress() {
+    if (upload.get_upload_status()) {
+        $("#compose-send-button").prop("disabled", true);
+    } else {
+        $("#compose-send-button").prop("disabled", false);
+    }
+}
```

Comment:
It would be a lot better if we didn't toggle the send button disabled/not-disabled on each check. What if this re-enables the send button when something else disabled it? 

Fixing the send button disable issue should be something that combines all of that logic to only enable the button if none of the reasons for disabling are true.

Reply:
Yeah, I'm still working on it. I currently have a function that has all of the logics for when to disable teh button, but I'm debating whether I should enable/disable the button in the function or should I just have the function return true/false and just let the call site handle the enabling/disabling.